modules/renter/streamer.go

```diff
@@ -0,0 +1,141 @@
+package renter
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"math"
+	"time"
+
+	"github.com/NebulousLabs/errors"
+)
+
+type (
+	// streamer is a io.ReadSeeker that can be used to stream downloads from
+	// the sia network.
+	streamer struct {
+		file   *file
+		offset int64
+		r      *Renter
+	}
+)
+
+// min is a helper function to find the minimum of multiple values.
+func min(values ...uint64) uint64 {
+	min := uint64(math.MaxUint64)
+	for _, v := range values {
+		if v < min {
+			min = v
+		}
+	}
+	return min
+}
+
+// Streamer create a io.ReadSeeker that can be used to stream downloads from
+// the sia network.
+func (r *Renter) Streamer(siaPath string) (string, io.ReadSeeker, error) {
+	// Lookup the file associated with the nickname.
+	lockID := r.mu.RLock()
+	file, exists := r.files[siaPath]
+	r.mu.RUnlock(lockID)
+	if !exists {
+		return "", nil, fmt.Errorf("no file with that path: %s", siaPath)
+	}
+	// Create the streamer
+	s := &streamer{
+		file: file,
+		r:    r,
+	}
+	return file.name, s, nil
+}
+
+// Read implements the standard Read interface. It will download the requested
+// data from the sia network and block until the download is complete.  To
+// prevent http.ServeContent from requesting too much data at once, Read can
+// only request a single chunk at once.
+func (s *streamer) Read(p []byte) (n int, err error) {
+	// Get the file's size
+	s.file.mu.RLock()
+	fileSize := int64(s.file.size)
+	s.file.mu.RUnlock()
+
+	// Make sure we haven't reached the EOF yet.
+	if s.offset >= fileSize {
+		return 0, io.EOF
+	}
+
+	// Calculate how much we can download. We never download more than a single chunk.
+	chunkSize := s.file.staticChunkSize()
+	remainingData := uint64(fileSize - s.offset)
+	requestedData := uint64(len(p))
+	remainingChunk := chunkSize - uint64(s.offset)%chunkSize
+	length := min(chunkSize, remainingData, requestedData, remainingChunk)
+
+	// Download data
+	buffer := bytes.NewBuffer([]byte{})
+	d, err := s.r.newDownload(downloadParams{
+		destination:       newDownloadDestinationWriteCloserFromWriter(buffer),
+		destinationType:   destinationTypeSeekStream,
+		destinationString: "httpresponse",
+		file:              s.file,
+
+		latencyTarget: 25e3 * time.Millisecond, // TODO high default until full latency suport is added.
+		length:        length,
+		needsMemory:   true,
+		offset:        uint64(s.offset),
+		overdrive:     3,  // TODO: moderate default until full overdrive support is added.
+		priority:      10, // TODO: high default until full priority support is added.
+	})
+	if err != nil {
+		return 0, errors.AddContext(err, "failed to create new download")
+	}
+
+	// Add the download object to the download queue.
+	// TODO: Maybe this is not necessary for streams?
+	s.r.downloadHistoryMu.Lock()
+	s.r.downloadHistory = append(s.r.downloadHistory, d)
+	s.r.downloadHistoryMu.Unlock()
```

Comment:
You are going to heavily clutter the download history if you append an item each time that `Read` is called. I would say that the best option is to add an item to `downloadHistory` when the `Streamer` function is called, and then otherwise we store the object in the `streamer` struct and just modify it to reflect that more data has been requested.

It is not typical for web browsers to add a streamed file to your download history (e.g. your download history doesn't extend if you watch a youtube video), which would be a solid argument towards not adding streaming items to the download history at all.

Since we're stuck with compatibility requirements if we add it, let's not add the streams to the download history for now, we can always add them in later if we decide that it's something that should be appearing in the download history.